### PR TITLE
feat(js): shared object modeling

### PR DESCRIPTION
**Pull request by genaris**
This PR: 

- Adds missing native binding methods for generic IndyCredx interface and nodejs implementation
- Adds all Indy CredX objects (subclasses of IndyObject) such as Schema, CredentialDefinition, etc. much in the same way as implemented in Python wrapper

### CODE REVIEW

File: wrappers/javascript/nodejs/src/NodeJSIndyCredx.ts [hunk 1]
```diff
@@ -60,11 +60,11 @@ export class NodeJSIndyCredx implements IndyCredx {
     return new ObjectHandle(ret.deref() as number)
   }
 
-  public schemaGetAttribute(options: { schema: ObjectHandle; name: string }) {
-    const { schema, name } = serializeArguments(options)
+  public schemaGetAttribute(options: { object: ObjectHandle; name: string }) {
+    const { object, name } = serializeArguments(options)
```
Comment (berendsliedrecht) [inline wrappers/javascript/nodejs/src/NodeJSIndyCredx.ts]: I think we should use the word `handle` or `objectHandle` here. `object` is also an internal keyword and might cause some annoying issues

File: wrappers/javascript/nodejs/src/NodeJSIndyCredx.ts [hunk 2]
```diff
@@ -565,18 +575,82 @@ export class NodeJSIndyCredx implements IndyCredx {
     return ret.deref() as string
   }
 
-  public presentationRequestFromJson(options: { json: string }) {
+  private objectFromJson(method: (byteBuffer: Buffer, ret: Buffer) => unknown, options: { json: string }) {
```
Comment (berendsliedrecht) [inline wrappers/javascript/nodejs/src/NodeJSIndyCredx.ts:578]: Oh that's nice. Lots of avoided repetition.

File: wrappers/javascript/shared/src/IndyCredx.ts [hunk 3]
```diff
@@ -3,26 +3,26 @@
 
 import type { ObjectHandle } from './ObjectHandle'
 
-export type CredentialEntry = {
+export type NativeCredentialEntry = {
```
Comment (berendsliedrecht) [inline wrappers/javascript/shared/src/IndyCredx.ts:6]: Any reason for using the `Native` prefix here?

**Review by berendsliedrecht — CHANGES_REQUESTED** [review]

**Comment by berendsliedrecht** [comment]
Great work @genaris!!! This takes so much load of my shoulders, really appreciated. 

When I know more about the Python wrapper I will go over it again, but we can merge this for now :).

File: wrappers/javascript/shared/src/IndyCredx.ts (hunk 3 above)
Comment (genaris) [reply wrappers/javascript/shared/src/IndyCredx.ts:6]: This naming was borrowed from Python wrapper when using types that include ObjectHandles (e.g. [RevocationConfig](https://github.com/animo/indy-shared-rs/blob/e152d08233a5dfa17f665554f09215793c668e17/wrappers/python/indy_credx/types.py#L667). IMHO I'm not convinced of this naming, just wanted to differentiate from the ones used in the recently defined API.

Comment (berendsliedrecht) [reply wrappers/javascript/shared/src/IndyCredx.ts:6]: Ah yeah okay. Again my lack of knowledge on the Python wrapper does not help here, but I would like to keep the wrapper as unopinionated as possible. So whatever indy-credx sends us, next to some casing and deserialisation changes I would not like to change things. If this does not happen I am okay with the naming like this or without the `Native` prefix.

**Pull request merged**